src/test/java/uk/gov/pay/ledger/event/resource/EventResourceAgreementIT.java

```diff
@@ -85,4 +76,45 @@ public void shouldWriteCancelledEvent() {
         assertThat(agreementEntity.getCancelledByUserEmail(), is("jdoe@example.org"));
         assertThat(agreementEntity.getCancelledDate(), is(ZonedDateTimeMatchers.within(1, ChronoUnit.SECONDS, now)));
     }
+
+    @Test
+    public void shouldNotWriteEventIfAMandatoryFieldIsNotPresent() {
```

Comment:
This doesn't cover fields other than event_type. We could add a few unit tests for other fields. 

A good place to add unit tests is https://github.com/alphagov/pay-ledger/blob/master/src/test/java/uk/gov/pay/ledger/event/resource/EventResourceTest.java#L24